mitxgraders/helpers/mitmath/calc.py

```diff
@@ -573,10 +574,39 @@ def handle_node(node):
 
             action = self.actions[node_name]
             handled_kids = [handle_node(k) for k in node]
-            return action(handled_kids)
 
-        # Find the value of the entire tree.
-        result = handle_node(self.tree)
+            # Check for nan
+            if any(np.isnan(item) for item in handled_kids if isinstance(item, float)):
+                return float('nan')
+
+            # Compute the result of this node
+            result = action(handled_kids)
+
+            # All actions convert the input to a number, array or IdentityMultiple
+            # Check if there were any infinities or nan
+            check = result
+            # IdentityMultiple needs to extract the value
+            if isinstance(result, IdentityMultiple):
```

Comment:
Currently, `IdentityMultiple(0)` returns integer `0`. You could just make `identityMultiple(float('nan'))` return `nan`. (It seems like that approach would work for other custom algebraic objects too, like dimensions if people were ever interested in that again.)

Reply:
What about `inf`?

Reply:
I need to extract `inf` in this manner anyway, I think, because otherwise we'll have floats adding to matrices, no?

Reply:
It would be ok if `inf` aborted a calculation, but we're not doing that...

Reply:
`IdentityMultiple(inf))` would need to stay `inf`, otherwise `1 + IdentityMultiple(inf)` would be scalar `inf`, which would be wrong.

Reply:
Oops. I understand. You extracted because isinf can't handle `IdentityMultiple`. Just ignore me :)

Reply:
I updated `IdentityMultiple` for `nan`, but I think I still need to do this check to catch `inf`.